examples/directml/phi/microsoft_phi.py

```diff
@@ -0,0 +1,96 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import argparse
+import json
+import shutil
+from pathlib import Path
+
+import onnxruntime as ort
+
+from olive.workflows import run as olive_run
+
+# pylint: disable=redefined-outer-name
+
+
+def optimize(model_name: str, optimized_model_dir: Path):
+    ort.set_default_logger_severity(4)
+    script_dir = Path(__file__).resolve().parent
+
+    model_info = {}
+
+    # Optimize the model with Olive
+    print(f"\nOptimizing {model_name}")
+
+    olive_config = None
+    with (script_dir / "config_phi.json").open() as fin:
+        olive_config = json.load(fin)
+
+    olive_config["input_model"]["config"]["model_path"] = model_name
+    olive_run(olive_config)
+
+    footprints_file_path = Path(__file__).resolve().parent / "gpu-dml_footprints.json"
+    with footprints_file_path.open("r") as footprint_file:
+        footprints = json.load(footprint_file)
+        conversion_footprint = None
+        for footprint in footprints.values():
+            if footprint["from_pass"] == "OnnxConversion":
+                conversion_footprint = footprint
+        assert conversion_footprint
+
+        model_info = {
+            "unoptimized": {
+                "path": Path(conversion_footprint["model_config"]["config"]["model_path"]),
+            },
+        }
+        print(f"Unoptimized Model : {model_info['unoptimized']['path']}")
+
+    # Create a copy of the unoptimized model directory, then overwrite with optimized models from the olive cache.
```

Comment:
We don't have the optimized models yet, so we can get rid of the second part of the sentence.